### ISSUE TITLE
Import of deprecated function triggers warning even if a non-deprecated macro of the same name is used

### ISSUE THREAD

**Issue by djc** [rust-lang/rust#148426]
I tried this code (with snapbox 0.6.23):

```rust
use snapbox::cmd::cargo_bin;

fn foo() {
    cargo_bin!();
}
```

There is:

- A deprecated function `snapbox::cmd::cargo_bin`
- A non-deprecated macro `snapbox::cmd::cargo_bin`

I expected to see this happen: no warning, since we're only using the macro.

Instead, this happened: import of the deprecated function triggered a warning.

```
error: use of deprecated function `snapbox::cmd::cargo_bin`: incompatible with a custom cargo build-dir, see instead `cmd::cargo_bin!`
 --> tests/suite/cli_rustup_init_ui.rs:5:29
  |
5 | use snapbox::cmd::{Command, cargo_bin};
  |                             ^^^^^^^^^
  |
  = note: `-D deprecated` implied by `-D warnings`
  = help: to override `-D warnings` add `#[allow(deprecated)]`
```

### Meta

`rustc --version --verbose`:
```
rustc 1.91.0 (f8297e351 2025-10-28)
binary: rustc
commit-hash: f8297e351a40c1439a467bbbb6879088047f50b3
commit-date: 2025-10-28
host: aarch64-apple-darwin
release: 1.91.0
LLVM version: 21.1.2
```

(Behavior is the same with rustc 1.93.0-nightly (b15a874aa 2025-11-02)).

**Comment by tingerrr** [comment]
For anyone having warnings as errors in their CI this a show-stopper, my current workaround is to use `#[expect(deprecated, reason = "...")]`, which should be fine as long as the upstream removal of one and deprecation of the other don't happen in the same dependency bump of the downstream project.
